Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -335,16 +335,20 @@ - (void)initSessionWithLaunchOptions:(NSDictionary *)options isReferrable:(BOOL)
     
     if ([BNCSystemObserver getOSVersion].integerValue >= 8) {
         if (![options objectForKey:UIApplicationLaunchOptionsURLKey] && ![options objectForKey:UIApplicationLaunchOptionsUserActivityDictionaryKey]) {
-            [self initUserSessionAndCallCallback:YES];
+            // If Facebook SDK is present, call deferred app link check here
+            if (![self checkFacebookAppLinks]) {
+                [self initUserSessionAndCallCallback:YES];
+            }
         }
         else if ([options objectForKey:UIApplicationLaunchOptionsUserActivityDictionaryKey]) {
-            self.preferenceHelper.isContinuingUserActivity = YES;
             if (self.accountForFacebookSDK) {
                 id activity = [[options objectForKey:UIApplicationLaunchOptionsUserActivityDictionaryKey] objectForKey:@"UIApplicationLaunchOptionsUserActivityKey"];
                 if (activity && [activity isKindOfClass:[NSUserActivity class]]) {
                     [self continueUserActivity:activity];
+                    return;
                 }
             }
+            self.preferenceHelper.shouldWaitForInit = YES;
```

Comment:
let's not move this line of code. I worked through a very large matrix of tests to make sure that I didn't break our init code. This would definitely break things.